sdks/go/pkg/beam/util/starcgenx/starcgenx.go

```diff
@@ -749,7 +749,7 @@ func (e *Extractor) varString(v *types.Var) string {
 // NameType turns a reflect.Type into a string based on it's name.
 // It prefixes Emit or Iter if the function satisfies the constrains of those types.
 func (e *Extractor) NameType(t types.Type) string {
-	switch a := t.(type) {
+	switch a := types.Unalias(t).(type) {
```

Comment:
This one doesn't need to change, since the default case already handled Aliases as desired AFAICT https://cs.opensource.google/go/go/+/refs/tags/go1.23.6:src/go/types/typestring.go;l=340